src/charm.py

```diff
@@ -450,6 +429,24 @@ def _common_exit_hook(self, update_ca_certs: bool = False) -> None:
             self.unit.status = BlockedStatus(str(e))
             return
 
+        # Update the catalogue entry here, as part of the common-exit-hook, because CatalogueItem
+        # doesn't take a callable for the URL.
+        # https://github.com/canonical/catalogue-k8s-operator/issues/41
+        # Which is totally ok, because we don't need this object for anything else.
+        CatalogueConsumer(
+            charm=self,
+            item=CatalogueItem(
+                name="Alertmanager",
+                icon="bell-alert",
+                url=self._external_url,
+                description=(
+                    "Alertmanager receives alerts from supporting applications, such as "
+                    "Prometheus or Loki, then deduplicates, groups and routes them to "
+                    "the configured receiver(s)."
+                ),
+            ),
+        )
+
```

Comment:
If reloading alertmanager fails, the item won't show up in the catalogue. Is that intentional? It seems like we'd want to list broken items as well, so that it is easy to discover that they indeed are broken.

Reply:
Only if the common exit hook _never_ ran to the end then we won't have a catalogue entry.
But that also means we'd have a bad status.

I could instead use the new lib from https://github.com/canonical/catalogue-k8s-operator/pull/42.

Reply:
Changed to use the update method from catalog v1.